File: src/Trackr/Task.cs
using System;
using System.Collections.Generic;

namespace Trackr
{
	public class Task
	{
		public string Id = Guid.NewGuid().ToString("N");
		public int Number;
		public string Title;
		public string Description;
		public string State;
		public IList<Comment> Comments = new List<Comment>();
	}
}

File: src/Trackr/app/model/Task.js
Ext.define('Trackr.model.Task', {
	extend: 'Ext.data.Model',
	requires: [ 'Trackr.data.proxy.Direct' ],
	fields: ['id', 'number', 'title', 'description', 'state'],
	hasMany: { model: 'Trackr.model.Comment', name: 'comments' },
	proxy: {
		type: 'direct2',
		// batchActions: false,
		paramOrder: ['id'], // Seems that are considered only for read operation
		api: {
			create: 'Server.TaskRepository.create',
			read: 'Server.TaskRepository.read',
			update: 'Server.TaskRepository.update',
			destroy: 'Server.TaskRepository.destroy'
		}
	}
});